GDJS/Runtime/bitmapfontmanager.js

```diff
@@ -74,43 +79,83 @@ gdjs.BitmapFontManager.prototype.removeFontUsed = function (name) {
   }
 };
 
-// This method is asynchronous and return a promise with the fontName.
-gdjs.BitmapFontManager.prototype.loadBitmapFont = async function (
+/**
+ * @param {string} bitmapFontResourceName
+ * @param {PIXI.Texture} texture
+ * @returns {PIXI.BitmapFont}
+ */
+gdjs.BitmapFontManager.prototype.getBitmapFontFromData = function (
   bitmapFontResourceName,
   texture
 ) {
+  // Reuse an existing bitmapFont that is already in memory and already installed
+  for (var i = 0; i < this._loadedFontsData.length; ++i) {
```

Comment:
Instead of an array, which is a pain to iterate on (and not efficient), use an object. The keys are the resource name. The values are the loadedFontData, which is an object that looks like `{ fontResourceFileName: string, fontData: string, fontName: string, }`.

When choosing a data structure, think about how to make access to the information you need as fast as possible.